tests/at_functional_test/test/pkam_verb_test.dart

```diff
@@ -0,0 +1,109 @@
+    var pkamResult = await read();
+    expect(pkamResult, 'data:success\n');
+  });
+
+  test('pkam authentication using the new syntax', () async {
+    await socket_writer(socketFirstAtsign!, 'from:$firstAtsign');
+    var fromResponse = await read();
+    print('from verb response : $fromResponse');
+    fromResponse = fromResponse.replaceAll('data:', '');
+    var pkamDigest = generatePKAMDigest(firstAtsign, fromResponse);
+    await socket_writer(socketFirstAtsign!,
+        'pkam:signingAlgo:rsa2048:hashingAlgo:sha256:$pkamDigest');
+    var pkamResult = await read();
+    expect(pkamResult, 'data:success\n');
+  });
+
+  test('pkam authentication - new syntax - passing invalid signing algo',
+      () async {
+    await socket_writer(socketFirstAtsign!, 'from:$firstAtsign');
+    var fromResponse = await read();
+    fromResponse = fromResponse.replaceAll('data:', '');
+    var pkamDigest = generatePKAMDigest(firstAtsign, fromResponse);
+    await socket_writer(socketFirstAtsign!,
+        'pkam:signingAlgo:rsa2047:hashingAlgo:sha256:$pkamDigest');
+    var pkamResult = await read();
+    expect(pkamResult.contains('Exception'), true);
+  });
+
+  test('pkam authentication using ecc ', () async {
+    try {
+      final eccAlgo = EccSigningAlgo();
+      var ec = getSecp256r1();
+      final eccPrivateKey = ec.generatePrivateKey();
+      eccAlgo.privateKey = eccPrivateKey;
+      // authenticating to the server to update the public key
+      await socket_writer(socketFirstAtsign!, 'from:$firstAtsign');
+      var fromResponse = await read();
+      fromResponse = fromResponse.replaceAll('data:', '');
+      var pkamDigest = generatePKAMDigest(firstAtsign, fromResponse);
+      await socket_writer(socketFirstAtsign!, 'pkam:$pkamDigest');
+      var pkamResponse = await read();
+      expect(pkamResponse, 'data:success\n');
+      // updating the public key to ecc public key
+      await socket_writer(socketFirstAtsign!,
+          'update:privatekey:at_pkam_publickey ${eccPrivateKey.publicKey.toString()}');
+      var response = await read();
+      expect(response, 'data:-1\n');
+      await socket_writer(socketFirstAtsign!, 'from:$firstAtsign');
+      fromResponse = await read();
+      fromResponse = fromResponse.replaceAll('data:', '');
+
+      final dataToSign = fromResponse.trim();
+      final dataInBytes = Uint8List.fromList(utf8.encode(dataToSign));
+      final signature = eccAlgo.sign(dataInBytes);
+      String encodedSignature = base64Encode(signature);
+      // var decodedSignature = base64Decode(encodedSignature);
+      // var verifyResult = eccAlgo.verify(dataInBytes, decodedSignature,
+      //     publicKey: eccPrivateKey.publicKey.toString());
+      await socket_writer(socketFirstAtsign!,
+          'pkam:signingAlgo:ecc_secp256r1:hashingAlgo:sha256:$encodedSignature');
+      var pkamResult = await read();
+      expect(pkamResult, 'data:success\n');
+    } finally {
```

Comment:
@purnimavenkatasubbu if there is an exception, we have to auth again and run the key update